src/veh_interact.cpp

```diff
@@ -897,131 +896,46 @@ void veh_interact::do_install()
                 install_info ) );
     install_info = std::make_unique<install_info_t>();
 
-    std::array<std::string, 8> &tab_list = install_info->tab_list = { {
-            pgettext( "Vehicle Parts|", "All" ),
-            pgettext( "Vehicle Parts|", "Cargo" ),
-            pgettext( "Vehicle Parts|", "Light" ),
-            pgettext( "Vehicle Parts|", "Util" ),
-            pgettext( "Vehicle Parts|", "Hull" ),
-            pgettext( "Vehicle Parts|", "Internal" ),
-            pgettext( "Vehicle Parts|", "Other" ),
-            pgettext( "Vehicle Parts|", "Filter" )
-        }
-    };
-
-    install_info->tab_list_short = { {
-            pgettext( "Vehicle Parts|", "A" ),
-            pgettext( "Vehicle Parts|", "C" ),
-            pgettext( "Vehicle Parts|", "L" ),
-            pgettext( "Vehicle Parts|", "U" ),
-            pgettext( "Vehicle Parts|", "H" ),
-            pgettext( "Vehicle Parts|", "I" ),
-            pgettext( "Vehicle Parts|", "O" ),
-            pgettext( "Vehicle Parts|", "F" )
-        }
-    };
+    std::vector<std::string> &tab_list = install_info->tab_list = {};
+    tab_list.push_back( pgettext( "Vehicle Parts|", "All" ) );
+    for( const std::string &cat : vpart_info::categories_all() ) {
+        tab_list.push_back( pgettext( "Vehicle Parts|", cat.c_str() ) );
```

Comment:
This is not a good way to use `pgettext`; it means that the relevant strings won't be picked up by the tool that extracts strings to be translated from the source code.  Probably the category names should be replaced by `translation` objects loaded from json.

Reply:
Can you expand on this a bit, I've read https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/TRANSLATING.md#developers but it shows usage of static
hardcoded translations or I don't think I understand what it's trying to say

I've categories as just plain strings right now, do I need to make a new
vpart_category_info, new json and deserialize that to use it?

Reply:
Based on your PR title, I was assuming that you were creating actual json objects to represent categories.  Now I think I misunderstood; you were simply making it possible to define categories of parts in json.

If you did want to go as far as making vpart categories json objects of their own, then these strings could be stored there.  That might be the best option in the long run.

Otherwise, there needs to be, somewhere in the source code, a call to `pgettext` (or `to_translation`) with literal string arguments representing the translation you want to do.

Reply:
Could you take a look at https://github.com/CleverRaven/Cataclysm-DDA/pull/44848/commits/8f38fccf2323076b57d3b5afa2c1d198fa4d299f ?
I'm loading categories from json via assign() into translation fields, do I need to do more to support localization?

Reply:
That looks promising.  A couple of things are still missing:
* You need to do is add some logic to `lang/extract_json_strings.py` to tell it which translatable fields need to be extracted from a `vehicle_part_category` json object.
* You should add the contexts to the strings in the json, so it would be e.g.
```
  {
    "id": "_all",
    "type": "vehicle_part_category",
    "name": { "ctxt": "Vehicle Parts|", "str": "All" },
    "short_name": { "ctxt": "Vehicle Parts|", "str": "*" },
    "//": "Hardcoded filter for any category.",
    "priority": 0
  },
```
(that's not a great context string, but matching the old one should mean the translators don't need to re-translate this stuff).